src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -1,398 +1,55 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Generic;
-using System.IO.Pipelines;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Connections;
-using Microsoft.AspNetCore.Connections.Experimental;
 using Microsoft.AspNetCore.Hosting.Server;
-using Microsoft.AspNetCore.Hosting.Server.Features;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal;
-using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
-using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core
 {
     public class KestrelServer : IServer
```

Comment:
Obsolete? This type is no longer used anywhere right?

Reply:
@halter73 ?

Reply:
I generally like the idea since having people who want to use Kestrel call UseKestrel and resolve the IServer is more future-proof given possible ctor changes. Removing KestrelServer entirely does break KestrelServer decorators though. Not sure how common those are.

This wouldn't be as big of an issue if we did something like what was suggested in #2350, but that's certainty not happening for .NET 5. If you try hard enough, you can get the needed functionality by modifying ServiceDescriptors as demonstrated by https://github.com/dotnet/aspnetcore/issues/2350#issuecomment-496506529.

I think given that KestrelServer decorators are probably uncommon, and a (complicated) workaround does exist, obsoleting is probably fine.

Reply:
I'll mark it as obsolete then 😄

Reply:
Do we need an API review on obsoleting types? @pranavkm

Reply:
You could shoot an email as an FYI.